content/en/docs/tasks/tls/manual-rotation-of-ca-certificates.md

```diff
@@ -0,0 +1,141 @@
+
+{{% /capture %}}
+
+{{% capture steps %}}
+## Choose an approach
+There are two ways to rotate or replace CA certificates.
+
+## Replace the cluster (recommended) {#replace-the-cluster}
+
+This is the safest approach to rotate or replace the CA certificates.
+
+1. Create a new cluster with new CA certificates.
+
+2. Migrate the workloads to the new cluster.
+
+3. Remove or destroy the old cluster.
+
+## Rotate the CA certificates manually
+
+{{< caution >}}
+Make sure to back up your certificate directory along with configuration files (.conf) and any other necessary files.
+
+Manually rotating the CA works by timing when a Kubernetes cluster knows about new certificates. Do not restart any pods or update any component in the cluster while manually rotating the CA.
+
+This approach requires a considerable downtime of your cluster.
+
+{{< /caution >}}
+
+1. Back up your `/etc/kubernetes` directory, referred to as the *Kubernetes conf directory* and `/etc/kubernetes/pki`, referred to as the *Kubernetes certificates directory*.
+
+
+1. Distribute new CA certificates (`ca.crt`, `ca.key`, `front-proxy-ca.crt` and `front-proxy-ca.key`) to all your control plane nodes in the Kubernetes certificates directory.
+
+
+    Copy the CA certificate file on all nodes needed by the kubelet and the control plane component's `--client-ca-file` flag in the manifest files. 
+
+    {{< note >}}
+
+    If you used `kubeadm` to deploy your cluster, the control plane components run as {{< glossary_tooltip text="static Pods" term_id="static-pod" >}}, managed by the {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}. Components are not disrupted by in-place updates of CA certificates until someone kills or restarts an already running Pod.
+
+    {{< /note >}}
+
+1. Get new certificates and private keys signed from the new CA for `apiserver.crt`, `apiserver-kubelet-client.crt`, and `front-proxy-client.crt` and replace them in the Kubernetes certificates directory.
+
+    To generate certificates and private keys for your cluster using the `openssl` command line tool, see [Certificates (`openssl`)](/docs/concepts/cluster-administration/certificates/#openssl).
+    
+    You can also use [`cfssl`](/docs/concepts/cluster-administration/certificates/#cfssl).
+
+
+1. If your cluster is using bootstrap tokens to join nodes, update the ConfigMap `cluster-info` in the `kube-public` namespace.
+
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    kubectl get cm/cluster-info --namespace kube-public -o yaml | \
+        /bin/sed "s/\(certificate-authority-data:\).*/\1 ${base64_encoded_ca}" | \
+        kubectl apply -f -
+    ```
+
+1. Update all service account tokens with the new CA certificates.
+    ```shell
+    base64_encoded_ca="$(base64 /etc/kubernetes/pki/ca.crt)"
+
+    for namespace in $(kubectl get ns --no-headers | awk '{print $1}'); do
```

Comment:
is this token signed by the older ca? If yes, just change the ca.crt will not be enough.
Also, if you delete the tokens it'll be recreated(with the ca used by the control-plane at the time)

Reply:
not until someone restart the control plane components in between which is not recommended.

Reply:
but just replace the ca will work?

Reply:
This is ok as written, as the guide is for rotating the CA, not the ServiceAccount signing keypair.

Reply:
Alternatively:
```suggestion
    for namespace in $(kubectl get namespace -o jsonpath='{.items[*].metadata.name}'); do
```